src/ome_zarr_models/__init__.py

```diff
@@ -1,6 +1,43 @@
 from importlib.metadata import PackageNotFoundError, version
 
+import zarr
+
+import ome_zarr_models.v04.hcs
+import ome_zarr_models.v04.image
+import ome_zarr_models.v04.image_label
+import ome_zarr_models.v04.labels
+import ome_zarr_models.v04.well
+from ome_zarr_models.base import BaseGroup
+from ome_zarr_models.v04.base import BaseGroupv04
+
 try:
     __version__ = version("ome_zarr_models")
 except PackageNotFoundError:  # pragma: no cover
     __version__ = "uninstalled"
+
+
+_V04_groups: list[type[BaseGroupv04]] = [
+    ome_zarr_models.v04.hcs.HCS,
+    ome_zarr_models.v04.image_label.ImageLabel,
+    ome_zarr_models.v04.image.Image,
+    ome_zarr_models.v04.labels.Labels,
+    ome_zarr_models.v04.well.Well,
+]
+
+
+def load_ome_zarr_group(group: zarr.Group) -> BaseGroup:
+    """
+    Create an ome-zarr-models object from an existing OME-Zarr group.
+
+    This function will 'guess' which type of OME-Zarr data exists by
+    trying to validate each group metadata definition against your data.
+    If validation is successful, that data class is returned without
+    trying any more.
+    """
+    for group_cls in _V04_groups:
```

Comment:
is `_V04_groups` ordered from most-to-least specific? Or is it otherwise impossible for a given zarr group to match two members of `_V04_groups`?

Reply:
This is just alphabetical, but any order is fine because every group has at least one required field, and none of the field names are shared between different groups.